Jetpack App (Basics): Update Control Activated Colors (Checkbox, Switch, Etc)

ParaskP7 opened the pull request:
Fixes #14436

This PR updates the `Jetpack`, but also `WordPress` (see exception), related control activated colors based on the latest design feedback.

## Jetpack App

Component/Theme | Before | After
-----|-----|-----
Checkbox/Light | <img width="250" height="500" alt="checkbox_light_before" src="https://user-images.githubusercontent.com/9729923/114885124-0a556a00-9e0f-11eb-83e1-6539af2600c5.png"> | <img width="250" height="500" alt="checkbox_light_after" src="https://user-images.githubusercontent.com/9729923/114887141-d5e2ad80-9e10-11eb-9e92-34dea5e3bafd.png">
Checkbox/Dark | <img width="250" height="500" alt="checkbox_dark_before" src="https://user-images.githubusercontent.com/9729923/114887211-e7c45080-9e10-11eb-9e67-cf04d613ea60.png"> | <img width="250" height="500" alt="checkbox_dark_after" src="https://user-images.githubusercontent.com/9729923/114887258-f3b01280-9e10-11eb-8863-bcb7f6906b04.png">
Switch/Light | <img width="250" height="500" alt="switch_light_before" src="https://user-images.githubusercontent.com/9729923/114887400-19d5b280-9e11-11eb-812d-7fdc63b8f850.png"> | <img width="250" height="500" alt="switch_light_after" src="https://user-images.githubusercontent.com/9729923/114887534-3671ea80-9e11-11eb-9b01-9c2f1ed45412.png">
Switch/Dark | <img width="250" height="500" alt="switch_dark_before" src="https://user-images.githubusercontent.com/9729923/114887625-4b4e7e00-9e11-11eb-9cee-ed7052f1945c.png"> | <img width="250" height="500" alt="switch_dark_after" src="https://user-images.githubusercontent.com/9729923/114887690-5a353080-9e11-11eb-830b-0469e8e20c31.png">

Notes:
- Jetpack App: You will notice that the `dark` theme is unaffected by the change.
- Jetpack App: You will notice that the `light` theme is changing from `primary_30` to `primary_50`, which is the default for this app.
- WordPress App: This PR doesn't affect the control activated colors for this app (minus the exception below).

## WordPress App

This is an extra update that happened as part of this PR where the below control activated colors where updated to the current WordPress app standard coloring system in terms of the control activated colors.

Screen/Component | Before | After
-----|-----|-----
Plugin Details/Switch | <img width="250" height="500" alt="plugin_details_switch_before" src="https://user-images.githubusercontent.com/9729923/114890631-dcbeef80-9e13-11eb-8b4d-7c13a9312369.png"> | <img width="250" height="500" alt="plugin_details_switch_after" src="https://user-images.githubusercontent.com/9729923/114890831-08da7080-9e14-11eb-8e3f-6782ebf2381f.png">
Number Picker Dialog/Switch | <img width="250" height="500" alt="number_picker_dialog_switch_before" src="https://user-images.githubusercontent.com/9729923/114891155-4fc86600-9e14-11eb-889e-e18d1231c84d.png"> | <img width="250" height="500" alt="number_picker_dialog_switch_after" src="https://user-images.githubusercontent.com/9729923/114891121-4b03b200-9e14-11eb-89f5-2df6ebc2f89f.png">

Notes:
- The `dark` theme is unaffected by the change.
- You will notice that the `light` theme is changing from `primary_40` to `primary_30`, which the control activated color for the `light` theme for the `WordPress` app across all components.

**To Test**
- Build, launch and smoke test the `blue` control activated colors (checkbox, switch, etc) for the `WordPress` app (e.g. `wordpressWasabiDebug`).
- Build, launch and smoke test the `green` control activated colors (checkbox, switch, etc) for the `Jetpack` app (e.g. `jetpackWasabiDebug`).

**To Test**
- Build, launch and smoke test the `white/blue` tab bar colors (text and line) for the `WordPress` app (e.g. `wordpressWasabiDebug`).
- Build, launch and smoke test the `white/green` tab bar colors (text and line) for the `Jetpack` app (e.g. `jetpackWasabiDebug`).

**Merge Instructions**
- ~~A quick design review by @osullivanchris on the tab color changes for the Jetpack app.~~
- ~~Remove `Needs Design Review` label.~~
- ~~A quick review of this previous PR (authored by Tyler Heck) on [Issue/update switch color #7857](https://github.com/wordpress-mobile/WordPress-Android/pull/7857). This is the PR where those switch theme exceptions appeared first.~~
- ~~Remove `Not Ready for Merge` label.~~
- Merge as normal.

## Regression Notes
1. Potential unintended areas of impact 🟢 
2. What I did to test those areas of impact (or what existing automated tests I relied on) 🟢 
3. What automated tests I added (or what prevented me from doing so) 🟢 

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ParaskP7 committed:
Feature: Remove the extra switch blue medium theme and its usages

This seems to be the only exception to the 'colorControlActivated' rule
that apply to every ui component within the codebase (like checkboxes,
switches, etc). As such, and in order to keep consistency across all the
components, without any exceptions whatsoever, this 'SwitchBlueMedium'
theme is now removed.

Note: This 'SwitchBlueMedium' exception was added as part of this PR:
- PR: Issue/update switch color #7857
- Link: https://github.com/wordpress-mobile/WordPress-Android/pull/7857

ParaskP7 committed:
Feature: Extract color control activated color to base colors

Since this 'colorControlActivated' color needs to be updated for the
'Jetpack' app, this commit extracts that to 'colors_base.xml' files and
updates the Jetpack specific light theme 'colorControlActivated' color
from the WordPress hardcoded 'primary_30' value to the Jetpack dynamic
'colorPrimary' color, just like it already happens on the dark them for
both apps.

ashiagr reviewed (approved):
Thanks @ParaskP7. Everything looks good code-wise. Also, I tested different screens with activated controls and found them to have expected colors. Good job 👍.

osullivanchris reviewed (approved):
My issues are resolved, thanks for the fix 👍  Generally everything looks good as well.

One thing on Jetpack. Smallest nitpick of all time, so I'm approving anyway.

If you look at the switches in Notifications settings, the light green colour used for the non circular part of the switch (I have drawn a grey line around it here). The colour is ever so slightly different on the master switch compared with the switch in the page. Its almost unnoticeable. But I just wonder is something hard coded there or not following the theme, as I see no reason the colours should be different. Maybe its just the way the master switch gets rendered for some reason.

<img width="949" alt="Screenshot 2021-04-16 at 15 01 39" src="https://user-images.githubusercontent.com/50576199/115036009-0421c500-9ec5-11eb-8801-7fd588c3a150.png">


> You will notice that the light theme is changing from primary_40 to primary_30, which the control activated color for the light theme for the WordPress app across all components.

Glad its consistent. Sounds like a fix. I have a question about whether this is the right colour globally, but I will bring this up elsewhere.

ParaskP7 commented:
👋 @osullivanchris !

> My issues are resolved, thanks for the fix 👍 Generally everything looks good as well.

🙏 

> One thing on Jetpack. Smallest nitpick of all time, so I'm approving anyway.

😅 🙏 

> If you look at the switches in Notifications settings, the light green colour used for the non circular part of the switch (I have drawn a grey line around it here). The colour is ever so slightly different on the master switch compared with the switch in the page. Its almost unnoticeable. But I just wonder is something hard coded there or not following the theme, as I see no reason the colours should be different. Maybe its just the way the master switch gets rendered for some reason.

Oh my, what an 👁️ for detail Chris, wow! 🥇 

I was checking the codebase and this exact master switch is actually a `androidx.appcompat.widget.SwitchCompat` component, which is different to the other switch, the one within the page, a `android.widget.Switch` component. This makes me think that the diff is due to this fact, but I haven't investigated deeper than that.

Do you want me to create a separate task and investigate this diff, to try and rectify it afterwards? In any case, I believe it is better if done separate as it will potentially end up being much more involved. Also, the same applies to the `WordPress` equivalent notification settings switch, right?

In any way, kudos for the finding! 🌟 

> Glad its consistent. Sounds like a fix. I have a question about whether this is the right colour globally, but I will bring this up elsewhere.

Yeap, thanks for that, I am seeing your comment "elsewhere" already. Thanks for trying to bring parity between the apps and color schemes! 🙏

osullivanchris commented:
> Oh my, what an 👁️ for detail Chris, wow! 🥇

Hahah I appreciate the recognition!!

> Do you want me to create a separate task and investigate this diff, to try and rectify it afterwards? In any case, I believe it is better if done separate as it will potentially end up being much more involved. Also, the same applies to the WordPress equivalent notification settings switch, right?

Yeah it does not need to be resolved. I wanted to call it out on the call side in case there was a component thing you wanted to improve. Like, should both use the same component? As you said it applies to WordPress as well. And it might be the lowest priority 'issue' of all time 😄

ParaskP7 commented:
👋 @osullivanchris !

> Hahah I appreciate the recognition!!

😄 

> Yeah it does not need to be resolved. I wanted to call it out on the call side in case there was a component thing you wanted to improve. Like, should both use the same component? As you said it applies to WordPress as well. And it might be the lowest priority 'issue' of all time 😄

I just created this #14484 issue, included that on the parent #14293 master issue, where we can further track this task there.

osullivanchris commented:
@ParaskP7 as discussed on slack with @mattmiklic, can we change the switches and checkboxes on WordPress to use the default colour as well? And then we no longer need an exception for Jetpack.

ParaskP7 commented:
👋 @osullivanchris !

> @ParaskP7 as discussed on slack with @mattmiklic, can we change the switches and checkboxes on WordPress to use the default colour as well? And then we no longer need an exception for Jetpack.

Sir yes sir, will have this ready for you to check by tomorrow! 🌟 Bye, bye miss exception pie!

JavonDavis commented:
Hey @ParaskP7 ! Bumping this to `17.3`, if you'd like to get it in for this release please merge this into release/17.2 and then DM me to release a new Beta version. It looks like it also might be ready soon so If you'd like to hold the first beta to get this in, please DM asap me to let me know. Thanks!

ParaskP7 commented:
👋 @JavonDavis !

> Hey @ParaskP7 ! Bumping this to 17.3, if you'd like to get it in for this release please merge this into release/17.2 and then DM me to release a new Beta version. It looks like it also might be ready soon so If you'd like to hold the first beta to get this in, please DM asap me to let me know. Thanks!

Definitely, we actually are not currently interested in having these changes released since we are targeting the `feature/jetpack-app-basics` feature branch. It is just that we have to add the milestone as it is part of the PR process (and anyway nice to have even for a feature branch).

Thanks for much for taking care of pumping this for us! 🙇

ParaskP7 committed:
Feature: Remove control activated color exception

It has been decided that both the 'WordPress' and 'Jetpack' app should
use the 'colorPrimary' for 'colorControlActivated'. As such, the control
activated color exception is no longer required.

ParaskP7 commented:
👋 @osullivanchris !

I have now made the update and removed the `primary_30` color control activated exception on the `WordPress/Light` variant. Below if the before/after screenshots.

Component/Theme | Before | After
-----|-----|-----
Checkbox/Light | <img width="250" height="500" alt="checkbox_light_before" src="https://user-images.githubusercontent.com/9729923/115365550-29684900-a1cd-11eb-9136-da97af334bd6.png"> | <img width="250" height="500" alt="checkbox_light_after" src="https://user-images.githubusercontent.com/9729923/115365594-32f1b100-a1cd-11eb-97ed-88f6431df768.png">
Switch/Light | <img width="250" height="500" alt="switch_light_before" src="https://user-images.githubusercontent.com/9729923/115365613-38e79200-a1cd-11eb-842a-60581b60df05.png"> | <img width="250" height="500" alt="switch_light_after" src="https://user-images.githubusercontent.com/9729923/115365633-3edd7300-a1cd-11eb-9acb-3bbfe381cf20.png">

Let me know if that looks okay to you. You can download the `WordPress` APK directly from this PR and play around with it yourself as well.

🙏 

PS: I am CCing you @mattmiklic as well since you where involved in the decision making.

osullivanchris commented:
> Let me know if that looks okay to you. You can download the WordPress APK directly from this PR and play around with it yourself as well.

Looks great, @ParaskP7. Thanks for the fix 🙌  One thing though - I tested in dark mode, and the checkboxes are still a different colour to the button there. Do you know what values we are using in dark mode? Do they go away from default?

ParaskP7 commented:
👋 @osullivanchris !

> Looks great, @ParaskP7. Thanks for the fix 🙌 

🙌 

> One thing though - I tested in dark mode, and the checkboxes are still a different colour to the button there. Do you know what values we are using in dark mode? Do they go away from default?

Yes, they go away from the default. 😃 

Actually, in `dark` mode, these values derived from the default even before us going through these changes, the only problem was on the `light` mode. With this change both, the `light` and `dark` mode are using the `colorPrimary`, while previous only the `dark` mode was using the `colorPrimary` and the light more using the hardcoded `primary_30`.

PS: The `colorPrimary` for WordPress is the `primary_50` for the `light` mode and the  `primary_30` for the `dark` mode.

osullivanchris commented:
Ok I think I follow! So you reckon dark mode, these components are using the correct default colour in WordPress (primary_30).

What I spotted is that they don't match the big button (restore site). So maybe the button is not using the correct values. Or maybe its ok. Maybe we use a different colour value for the buttons. I know that white text would be hard to read on blue30

ParaskP7 commented:
👋 @osullivanchris !

> Ok I think I follow! So you reckon dark mode, these components are using the correct default colour in WordPress (primary_30).

Actually no! 😄 This is a `Jetpack.PrimaryButton`, which has its parent as `WordPress.Button.Primary`, which on the light mode has a background of `@color/secondary_disabled_selector`, while on the dark more a background of `@color/secondary_variant_disabled_selector`, notice the `variant` diff here.

Now this dark more `@color/secondary_variant_disabled_selector` selector is using the `colorSecondaryVariant`, which for the WordPress dark mode is the `primary_50`, which is not matching the `colorControlActivated` as you observing. So, it is not the `primary_30` as you would expect, but rather the `primary_50` in this case. 

> What I spotted is that they don't match the big button (restore site). So maybe the button is not using the correct values. Or maybe its ok. Maybe we use a different colour value for the buttons. I know that white text would be hard to read on blue30.

As per the above, you actually spotted it correctly once more! 🥇 

PS: The light more `@color/secondary_disabled_selector` selector is using the `colorSecondary`, which for the WordPress dark mode is the `primary_50`, which is exactly what we want and match that of the `colorControlActivated`.
PSS: Just as an FYI, nowadays the `colorSecondaryVariant` is the same as `colorPrimaryVariant`, just like `colorSecondary` is the same as `colorPrimary`, so please don't be confused by the `secondary` vs `primary` naming convention as both point to the same colors.

ParaskP7 commented:
👋 @osullivanchris !

> Actually no! 😄 This is a Jetpack.PrimaryButton, which has its parent as WordPress.Button.Primary, which on the light mode has a background of @color/secondary_disabled_selector, while on the dark more a background of @color/secondary_variant_disabled_selector, notice the variant diff here.

As discussed, I just created this #14502 issue, included that on the parent #14293 master issue, where we can further track this task there.
